src/twisted/web/http.py

```diff
@@ -2431,14 +2434,17 @@ def allContentReceived(self):
 
         self._handlingRequest = True
 
+        # We go into raw mode here even though we will be receiving lines next
+        # in the protocol; however, this data will be buffered and then passed
+        # back to line mode in the setLineMode call in requestDone.
+        self.setRawMode()
```

Comment:
Maybe  add some note about pipelining to help other understand what is going on here.

```suggestion
        # back to line mode in the setLineMode call in requestDone.
        # This is done to comply with the HTTP pipelining specification.
        # https://en.wikipedia.org/wiki/HTTP_pipelining
        self.setRawMode()
```

Reply:
I don't find this reference particularly illuminating, since we specifically *aren't* doing pipelining here, just buffering